### PR TITLE
docs: improve documentation and add #[warn(missing_docs)]

### LINKED ISSUE #65: Missing docstrings
Most of the sources are lacking docstrings.
It would be useful to add short comments for each function/struct.

### PR DESCRIPTION
**Pull request by joshuachp**
Closes #65

### CODE REVIEW

**Comment by joshuachp** [comment]
This PR is based on the work of #60. Which can be closed

File: src/data/mock_astarte.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/*
```
Comment (harlem88) [inline src/data/mock_astarte.rs:1]: This file was renamed to `mock_astarte_sdk.rs` in other PR, delete it.

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/data/mock_astarte.rs (hunk 1 above)
Comment (joshuachp) [reply src/data/mock_astarte.rs]: Donte 👍🏻

File: README.md [hunk 2]
```diff
@@ -12,11 +12,21 @@
...
 
+## Requirements
+
+- protobuf >= 3.15
+- Rust version >= 1.59
+
+For more information see:
+
+- [ARCHITECTURE.md](./docs/ARCHITECTURE.md)
+- [Astarte Documentation](https://docs.astarte-platform.org/latest/001-intro_user.html)
```
Comment (harlem88) [inline README.md:23]:
```suggestion
## Documentation:

- [Astarte Message Hub Documentation](https://docs.rs/astarte-message-hub/0.1.0/astarte_message_hub)
- [ARCHITECTURE.md](./docs/ARCHITECTURE.md)
- [Astarte Documentation](https://docs.astarte-platform.org/latest/001-intro_user.html)

## Requirements

- protobuf >= 3.15
- Rust version >= 1.59
```

File: README.md [hunk 3]
```diff
@@ -45,218 +55,3 @@ store_directory = "<STORE_PAHT>"
 
 An example configuration file can be found in the [examples](./examples/message-hub-config.toml)
 direction.
```
Comment (harlem88) [inline README.md:57]: Add a reference to the example.
```
## Example
Have a look at the [examples](./examples/) for an usage example showing how to send and receive data.

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -12,11 +12,22 @@
 A central service that runs on (Linux) devices for collecting and delivering messages from N apps
 using 1 MQTT connection to Astarte.
 
+## Documentation
+
+- [Astarte Message Hub Documentation](https://docs.rs/astarte-message-hub/0.1.0/astarte_message_hub)
```
Comment (harlem88) [inline README.md:17]: Leave only astarte-message-hub for the latest version
```suggestion
- [Astarte Message Hub Documentation](https://docs.rs/astarte-message-hub)
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

**Review by harlem88 — APPROVED** [review]

**Review by harlem88 — APPROVED** [review]

**Pull request merged**